compiler/rustc_infer/src/infer/mod.rs

```diff
@@ -721,6 +744,29 @@ impl<'tcx> InferCtxt<'tcx> {
         freshen::TypeFreshener::new(self, true)
     }
 
+    /// Effect fallback is infallible. We always figure out a concrete value for the effect
+    /// from the environment. If we can't figure out something we fall back to the effect
+    /// not applying, which is always a safe choice.
+    pub fn effects_inference_fallback(&self) {
+        let mut inner = self.inner.borrow_mut();
+        let mut table = inner.effect_unification_table();
+        let n = table.len();
+
+        for i in 0..n {
+            let index = i as u32;
+            let vid = ty::EffectVid { index, phantom: PhantomData };
+            let EffectVarValue { val, origin } = table.probe_value(vid);
+            match val {
+                EffectVariableValue::Unknown { .. } => {}
+                EffectVariableValue::Known { .. } => continue,
+            }
+            let value = self.tcx.mk_effect(ty::EffectValue::Rigid { on: true }, origin.effect_kind);
```

Comment:
not applying vs `ty::EffectValue::Rigid { on: true },`. afaict either the function comment or the impl is wrong